File: modules/losowe.py
import discord
from discord.ext import commands
import requests
from bs4 import BeautifulSoup
import re
import time

class RandomMemeCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.listener()
    async def on_ready(self):
        print(f"Logged in as {self.bot.user}")

    @commands.command()
    async def losowe(self, ctx):
        max_retries = 5
        for _ in range(max_retries):
            try:
                response = requests.get("https://jbzd.com.pl/losowe")
                soup = BeautifulSoup(response.content, 'html.parser')

                pattern = re.compile(r'https://i1\.jbzd\.com\.pl/contents/\d{4}/\d{2}/[a-zA-Z0-9]+\.(jpg|gif|png|mp4)')
                meme_element = soup.find(lambda tag: tag.name in ['img', 'video'] and 'src' in tag.attrs and pattern.match(tag['src']))
                meme_url = meme_element['src'] if meme_element else None

                title_element = soup.find('meta', {'property': 'og:title'})
                title = title_element['content'] if title_element else "Random Meme"

                if meme_url:
                    embed = discord.Embed(title=title)
                    if meme_url.endswith('.mp4'):
                        embed.add_field(name="Meme", value=meme_url)
                    else:
                        embed.set_image(url=meme_url)
                    await ctx.send(embed=embed)
                    return  # Meme found, exit the function

            except Exception as e:
                print(f"Error fetching meme: {e}")  # Print the error for debugging

            time.sleep(1)  # Optional: Sleep for a short time before retrying

        # If no meme is found after all retries
        await ctx.send("Unable to find a meme after several attempts.")

async def setup(bot):
    await bot.add_cog(RandomMemeCog(bot))

File: modules/losowe.js
const { SlashCommandBuilder, EmbedBuilder } = require('discord.js');

async function fetchMeme() {
  try {
    const res = await fetch('https://jbzd.com.pl/losowe');
    const html = await res.text();
    const pattern = /https:\/\/i1\.jbzd\.com\.pl\/contents\/\d{4}\/\d{2}\/[^"']+\.(jpg|gif|png|mp4)/;
    const match = html.match(pattern);
    const titleMatch = html.match(/<meta property="og:title" content="([^"]+)"/);
    return { url: match ? match[0] : null, title: titleMatch ? titleMatch[1] : 'Random Meme' };
  } catch (e) {
    return { url: null, title: 'Random Meme' };
  }
}

module.exports = {
  data: new SlashCommandBuilder()
    .setName('losowe')
    .setDescription('Random meme from jbzd.com.pl'),
  async execute(interaction) {
    const { url, title } = await fetchMeme();
    if (!url) {
      await interaction.reply('Unable to find a meme.');
      return;
    }
    const embed = new EmbedBuilder().setTitle(title);
    if (url.endsWith('.mp4')) {
      embed.setDescription(url);
    } else {
      embed.setImage(url);
    }
    await interaction.reply({ embeds: [embed] });
  }
};
